src/psyclone/psyir/transformations/loop_fuse_trans.py

```diff
@@ -118,6 +117,13 @@ def validate(self, node1, node2, options=None):
             raise TransformationError(
                 f"Error in {self.name} transformation. Nodes are not siblings "
                 f"who are next to each other.")
+
+        # Check node1 comes before node2:
+        if node2.position-node1.position != 1:
+            raise TransformationError(
+                f"Error in {self.name} transformation. The second loop comes "
```

Comment:
The check is more general than the error message says. Perhaps: "the second loop does not immediately follow the first loop"?